.devcontainer/swift/user.sh

```diff
@@ -0,0 +1,22 @@
+set -xe
+
+# add the workspace to the codeql search path
+mkdir -p /home/vscode/.config/codeql
+echo "--search-path /workspaces/codeql" > /home/vscode/.config/codeql/config
+
+# create a swift extractor pack with the current state
+cd /workspaces/codeql
+bazel run swift/create-extractor-pack
+
+#install and set up pre-commit
+python3 -m pip install pre-commit --no-warn-script-location
+$HOME/.local/bin/pre-commit install
+
+cat >> $HOME/.bashrc << EOF
+
+# have the codeql binary installed by vscode automatically in PATH
+bin=\$(ls \$HOME/.vscode-remote/data/User/globalStorage/github.vscode-codeql/*/codeql/codeql -t 2>/dev/null | head -1)
```

Comment:
Does this work? I.e. is the codeql vscode extension initialized before this script is run?

Reply:
no, and in any case user intervention is required any way to download the codeql CLI. So this works only after doing that and eventually restarting the terminal (not the codespace connection). Do you think there is a more user-friendly solution? Should we bypass the codeql CLI installation doing it in the codespace setup?

Reply:
I missed the part with it being added to bashrc.
That means that zsh users (like me) don't get this at all.

I'd rather just download the latest CLI release and put it on the PATH - vscode can be configured to use it as well with a setting you can set from the devcontainer.

Reply:
(arguably, then zsh users still need to set up their PATH manually, but it'd be a bit easier)

Reply:
I've ended up putting the auto-installation in `root.sh`, with a symbolic link in `/usr/local/bin` that will then be picked up automatically by both the vscode plugin and zsh :slightly_smiling_face: 

And a handy `update-codeql` command is also thrown in in the bucket.

Reply:
perfect!